src/Documentation/Grain-Versioning/Grain-versioning.md

```diff
@@ -0,0 +1,35 @@
+# Grain versioning
+
+> [!WARNING]
+> This page describes how to use grain interface versioning. The versioning of
+> Grain state it out of the scope.
+
+## Overview
+On a given cluster, silos can support different versions of a grain type.
+![Cluster with different versions of a grain](version.png)
+In this example the client and Silo{1,2,3} were compiled with grain interface `A` version 1. Silo 4 was compiled with `A` version 2.
+
+## Limitations:
+-	No versioning on stateless worker
+-	Streaming agents are not versioned
+
+## Enable versioning
+By default, grains are not versioned. You can version grain by using the VersionAttribute on the grain interface:
+
+``` cs
+[Version(X)]
+public interface IVersionUpgradeTestGrain : IGrainWithIntegerKey {}
+```
+
+Where `X` is the version of the grain interface.
+
+## Grain version compatibility and placement
+When a call from a versioned grain arrive in a cluster:
+-	If no activation exists, a compatible activation will be created
+-	If an activation exists:
+ - If the current one is not compatible, it will be deactivated and new compatible will be created
+ - If the current one is compatible, the call will be handled normally
```

Comment:
indenting is off. Please don't use tabs here, but add spaces before the `-` for nested lists